samtools.wdl

```diff
@@ -23,29 +24,29 @@ task BgzipAndIndex {
     }
 
     runtime {
-        docker: "quay.io/biocontainers/tabix:" + dockerTag
+       docker: dockerImage
     }
 }
 
 task Index {
     input {
         File bamFile
-        String bamIndexPath
-
-        String dockerTag = "1.8--h46bd0b3_5"
+        String dockerImage = "quay.io/biocontainers/samtools:1.8--h46bd0b3_5"
     }
+    String bamPath = basename(bamFile)
 
     command {
-        samtools index ~{bamFile} ~{bamIndexPath}
+        ln ~{bamFile} ~{bamPath}
+        samtools index ~{bamPath}
     }
 
     output {
-        File indexedBam = bamFile
-        File index = bamIndexPath
+        File indexedBam = bamPath
+        File index =  sub(bamPath, "\.bam$", ".bai")
```

Comment:
I think this would actually result in `bamPath + ".bai"`.

Reply:
This should result in `bla.bam` -> `bla.bai` it is the same as used in the alignment tasks.

Reply:
Whoops you are correct. Samtools index creates `.bam.bai` files. 
I have adapted the task to create a `.bai` file regardless.